contracts/staking-vault/StakingVault.sol

```diff
@@ -0,0 +1,284 @@
+pragma solidity ^0.8.0;
+
+import "@openzeppelin-4.8.0/token/ERC1155/ERC1155.sol";
+import "@openzeppelin-4.8.0/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin-4.8.0/token/ERC20/extensions/IERC20Metadata.sol";
+
+import "../../interfaces/IPriceFeeds.sol";
+import "../proxies/0_8/Upgradeable_0_8.sol";
+
+contract StakingVault is Upgradeable_0_8, ERC1155 {
+  using SafeERC20 for IERC20;
+  struct TokenComposition {
+    address depositToken;
+    address tokenToBack;
+  }
+  address public protocol;
+  IPriceFeeds public priceFeed;
+  address public valuationToken;
+  address[] public stakingTokens;
+  mapping(address => bool) public tokenSupported;
+  mapping(uint256 => uint256) public sTokenPrice;
+  mapping(uint256 => TokenComposition) public identifierToTokens;
+  mapping(uint256 => uint256) public balanceStakedPerID;
```

Comment:
this stores same amount as if uint256->bytes32 right? technically its abetter collision protection than `uint256(keccak256(abi.encode(depositToken, tokenToBack)));`

Reply:
ok I got why you're doing int. because ERC1155 standard